### PR TITLE
reuse provided site instance if available

### PR DESCRIPTION
**Pull request by mojavelinux**
- addresses Jekyll core issue https://github.com/jekyll/jekyll/issues/4858

### CODE REVIEW

File: lib/jekyll/watcher.rb [hunk 1]
```diff
@@ -4,10 +4,10 @@ module Jekyll
   module Watcher
     extend self
 
-    def watch(options)
+    def watch(options, site = nil)
```
Comment (parkr) [inline lib/jekyll/watcher.rb]: Wonderful! Would you mind adding a comment that the `site` is only populated in Jekyll v3.2 (upcoming) and up? Thanks!

**Comment by parkr** [comment]
LGTM! Just one thought above.

File: lib/jekyll/watcher.rb (hunk 1 above)
Comment (mojavelinux) [inline lib/jekyll/watcher.rb]: Gladly!

**Comment by mojavelinux** [comment]
API documentation added!

**Comment by parkr** [comment]
LGTM! Thank you.

**Comment by envygeeks** [comment]
LGTM.

**Comment by parkr** [comment]
@jekyllbot: merge

**Comment by parkr** [comment]
released in v1.5.0!

**Pull request merged**